File: Actividad 08-11-2024/Main.py
#Clase base
class Animal:
    def __init__(self, edad):
        self.edad = edad
        self._color = None #Atributo privado

    def hacer_sonido(self):
        print("El animal hace un sonido")

    def comer(self):
        print("El animal esta comiendo")

    def set_color(self, color):
        self._color = color

    def get_color(self):
        return self._color

#Clase derivada
class Perro(Animal):
    def hacer_sonido(self):
        print("El perro ladra")

    def comer(self):
        print("El perro esta comiendo croquetas")

#Clase derivada
class Gato(Animal):
    def hacer_sonido(self):
        print("El gato maulla")

    def comer(self):
        print("El gato esta comiendo pescado")

#Clase derivada
class Pajaro(Animal):
    def hacer_sonido(self):
        print("El pajaro canta")

    def comer(self):
        print("El pajaro esta comiendo alpiste")

#Funcion principal
def main():
    miAnimal = Animal(5)
    miAnimal.hacer_sonido()
    miAnimal.comer()

    miPerro = Perro(7)
    miPerro.hacer_sonido()
    miPerro.comer()
    miPerro.set_color("Marron")
    print("Color del perro: ", miPerro.get_color())

    miGato = Gato(3)
    miGato.hacer_sonido()
    miGato.comer()
    miGato.set_color("Blanco")
    print("Color del gato: ", miGato.get_color())

    miPajaro = Pajaro(1)
    miPajaro.hacer_sonido()
    miPajaro.comer()
    miPajaro.set_color("Amarillo")
    print("Color del pajaro: ", miPajaro.get_color())

if __name__ == "__main__":
    main()

File: Actividad 08-11-2024/Main.java
// Clase base
class Animal{
    protected int edad;
    private String color;
    public Animal (int e){
        edad = e;
    }
    public void hacerSonido(){
        System.out.println("El animal hace un sonido");
    }
    public void comer(){
        System.out.println("El animal esta comiendo");
    }
    public void setColor(String color){
        this.color = color;
    }
    public String getColor(){
        return color;
    }
}

// Clase derivada
class Perro extends Animal{
    public Perro(int e){
        super(e);
    }
    public void setEdad (int e){
        edad = e;
    }
    @Override
    public void hacerSonido(){
        System.out.println("El perro ladra");
    }
    @Override
    public void comer(){
        System.out.println("El perro esta comiendo croquetas");
    }
}

//Clase derivada
class Gato extends Animal{
    public Gato(int e){
        super(e);
    }
    public void setEdad (int e){
        edad = e;
    }
    @Override
    public void hacerSonido(){
        System.out.println("El gato maulla");
    }
    @Override
    public void comer(){
        System.out.println("El gato esta comiendo pescado");
    }
}

//Clase derivada
class Pajaro extends Animal{
    public Pajaro(int e){
        super(e);
    }
    public void setEdad (int e){
        edad = e;
    }
    @Override
    public void hacerSonido(){
        System.out.println("El pajaro canta");
    }
    @Override
    public void comer(){
        System.out.println("El pajaro esta comiendo alpiste");
    }
}

public class Main{
    public static void main(String[] args){
        Animal miAnimal = new Animal(5);
        miAnimal.hacerSonido();
        miAnimal.comer();
        
        Perro miPerro = new Perro(7);
        miPerro.hacerSonido();
        miPerro.comer();
        miPerro.setColor("Marron");
        System.out.println("Color del perro: " + miPerro.getColor());
        
        Gato miGato = new Gato(3);
        miGato.hacerSonido();
        miGato.comer();
        miGato.setColor("Blanco");
        System.out.println("Color del gato: " + miGato.getColor());
        
        Pajaro miPajaro = new Pajaro(1);
        miPajaro.hacerSonido();
        miPajaro.comer();
        miPajaro.setColor("Amarillo");
        System.out.println("Color del pajaro " + miPajaro.getColor());
    }
}